src/System.Runtime.Numerics/src/System/Numerics/Complex.cs

```diff
@@ -120,6 +191,51 @@ public static Complex Divide(Complex dividend, Complex divisor)
             }
         }
 
+        public static Complex operator /(Complex left, double right)
```

Comment:
We can reduce the duplicated checks by doing:
```C#
if (right == 0)
{
    return new Complex(double.NaN, double.NaN);
}

if (!double.IsFinite(left.m_real))`
{
    if (!double.IsFinite(left.m_imaginary))
    {
        return new Complex(double.NaN / right, double.NaN);
    }

    return new Complex(left.m_real / right, double.NaN);
}

if (!double.IsFinite(left.m_imaginary))
{
    return new Complex(double.NaN, left.m_imaginary / right);
}

return new Complex(left.m_real / right, left.m_imaginary / right);
```

Reply:
Also, adding comments on why we return `NaN` for each of these would be helpful, that way people don't need to go digging into the underlying math to determine the why for each scenario.

Reply:
@tannergooding These comments addressed.